adapters/teads/teads.go

```diff
@@ -0,0 +1,178 @@
+			}
+		}
+
+		var defaultImpExt DefaultBidderImpExtension
+		if err := json.Unmarshal(imp.Ext, &defaultImpExt); err != nil {
+			return &errortypes.BadInput{
+				Message: "Error parsing Imp.Ext object",
+			}
+		}
+		imp.TagID = strconv.Itoa(defaultImpExt.Bidder.PlacementId)
+		teadsImpExt := &TeadsImpExtension{
+			KV: TeadsKV{
+				PlacementId: defaultImpExt.Bidder.PlacementId,
+			},
+		}
+		if extJson, err := json.Marshal(teadsImpExt); err != nil {
+			return &errortypes.BadInput{
+				Message: "Error stringify Imp.Ext object",
+			}
+		} else {
+			imp.Ext = extJson
+		}
+	}
+	return nil
+}
+
+// Builds enpoint url based on adapter-specific pub settings from imp.ext
+func (a *adapter) buildEndpointURL() (string, error) {
+	endpointParams := macros.EndpointTemplateParams{}
+	host, err := macros.ResolveMacros(a.endpointTemplate, endpointParams)
+
+	if err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Unable to parse endpoint url template: " + err.Error(),
+		}
+	}
+
+	thisURI, err := url.Parse(host)
+	if err != nil {
+		return "", &errortypes.BadInput{
+			Message: "Malformed URL: " + err.Error(),
+		}
+	}
+
+	return thisURI.String(), nil
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, _ *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{fmt.Errorf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}
+	}
+
+	var bidResp openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(bidResp.SeatBid))
+
+	for _, sb := range bidResp.SeatBid {
+		for i := 0; i < len(sb.Bid); i++ {
+			bid := sb.Bid[i]
+
+			var bidExtTeads TeadsBidExt
+			if err := json.Unmarshal(bid.Ext, &bidExtTeads); err != nil {
+				return nil, []error{err}
+			}
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid: &bid,
+				BidMeta: &openrtb_ext.ExtBidPrebidMeta{
+					RendererName:    bidExtTeads.Prebid.Meta.RendererName,
+					RendererVersion: bidExtTeads.Prebid.Meta.RendererVersion,
+				},
+				BidType: getMediaTypeForImp(bid.ImpID, internalRequest.Imp),
+			})
+		}
+	}
+	if bidResp.Cur != "" {
+		bidResponse.Currency = bidResp.Cur
+	}
+	return bidResponse, []error{}
+}
+
+func getMediaTypeForImp(impID string, imps []openrtb2.Imp) openrtb_ext.BidType {
+	for _, imp := range imps {
+		if imp.ID == impID {
+			if imp.Video != nil {
+				return openrtb_ext.BidTypeVideo
+			} else if imp.Native != nil {
+				return openrtb_ext.BidTypeNative
+			}
```

Comment:
https://github.com/prebid/prebid-server/blob/0256b2a58170f7fbc11262001db3c7ee752348c3/static/bidder-info/teads.yaml#L7-L9

As per bidder-info yaml, native media format is not supported by `Teads` adapter. 
@github-maxime-liege could you confirm whether `Teads` adapter only supports banner and video. If yes then should update adpater code to remove native check on line 172